src/Microsoft.Azure.SignalR.Protocols/CheckWithAckMessage.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.SignalR.Protocol
+{
+    /// <summary>
+    /// Base class of check-with-ack messages between Azure SignalR Service and SDK.
+    /// </summary>
+    public abstract class CheckWithAckMessage : ExtensibleServiceMessage, IAckableMessage, IMessageWithTracingId
+    {
+        /// <summary>
+        /// Gets or sets the ack id.
+        /// </summary>
+        public int AckId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tracing Id
+        /// </summary>
+        public ulong? TracingId { get; set; }
+
+        protected CheckWithAckMessage(int ackId, ulong? tracingId)
+        {
+            AckId = ackId;
+            TracingId = tracingId;
+        }
+    }
+
+    /// <summary>
+    /// A waiting for ack check-user-in-group message.
+    /// </summary>
+    public class CheckUserInGroupWithAckMessage : CheckWithAckMessage
+    {
+        /// <summary>
+        /// Gets or sets the user Id.
+        /// </summary>
+        public string UserId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the group name.
+        /// </summary>
+        public string GroupName { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LeaveGroupWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="userId">The connection Id.</param>
```

Comment:
Why the name is `userId` and the description is "connection Id"?

Reply:
nice catch, just move the old one https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR.Protocols/GroupMessage.cs#L279 to here